File: app/src/main/java/com/person/v_plaunov/mylistview/CoinViewActivity.java
package com.person.v_plaunov.mylistview;

import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.graphics.drawable.Drawable;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.TextView;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

public class CoinViewActivity extends AppCompatActivity {

    //private static String DB_PATH = "/data/data/com.person.v_plaunov.mylistview/databases/";
    //private static String DB_PATH = "/storage/self/primary/My Documents/";
    private static String DB_NAME = "myDB";
    private SQLiteDatabase myDataBase;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_coinview);

        Intent intent = getIntent();
        Bundle bundle = intent.getExtras();
        int coinId = bundle.getInt("coin_id");
//        String coinState = bundle.getString("coin_state");
//        String coinNominal = bundle.getString("coin_nominal");
//        String coinImgName = bundle.getString("coin_img");

        // переменные для query
        String [] columns = {"_id", "Nominal", "State", "Img", "Year", "Theme", "Description"};
        String selection;
        String[] selectionArgs;

        //coins.clear();
        //Наш ключевой хелпер
        DBOpenHelper dbOpenHelper = new DBOpenHelper(this, DB_NAME);
        myDataBase = dbOpenHelper.openDataBase();
        //Все, база открыта!

//        Cursor cursor = myDataBase.query("coins", columns, selection, selectionArgs, null, null, null);
        selection = "_id = ?";
        selectionArgs = new String[] {String.valueOf(coinId)};
        Cursor cursor = myDataBase.query("coins", columns, selection, selectionArgs, null, null, null);
        String coinImagePath = "";
        cursor.moveToFirst();
        String coinState = cursor.getString(cursor.getColumnIndexOrThrow("State"));
        TextView cState = findViewById(R.id.coin_state);
        cState.setText(coinState);

        String coinNominal = cursor.getString(cursor.getColumnIndexOrThrow("Nominal"));
        TextView cNominal = findViewById(R.id.coin_nominal);
        cNominal.setText(coinNominal);
        String coinImgName = cursor.getString(cursor.getColumnIndexOrThrow("Img"));
        if (!TextUtils.isEmpty(coinImgName))
            coinImgName = coinImgName.replace("\\", "/");
        String coinYear = cursor.getString(cursor.getColumnIndexOrThrow("Year"));
        TextView cYear = findViewById(R.id.coin_year);
        cYear.setText(coinYear);
        String coinTheme = cursor.getString(cursor.getColumnIndexOrThrow("Theme"));
        TextView cTheme = findViewById(R.id.coin_theme);
        cTheme.setText(coinTheme);
        String coinDescr = cursor.getString(cursor.getColumnIndexOrThrow("Description"));
        TextView cDescr = findViewById(R.id.coin_description);
        cDescr.setText(coinDescr);
        cursor.close();

        String packageName = this.getPackageName();
        String DB_PATH = String.format(this.getString(R.string.str_db_path), packageName);
        if (!TextUtils.isEmpty(coinImgName))
            coinImagePath = DB_PATH + coinImgName;

        try {
            InputStream inputStream = new FileInputStream(coinImagePath);
            Drawable d = Drawable.createFromStream(inputStream, null);
            TouchImageView cImg = findViewById(R.id.coin_pic);
            cImg.setImageDrawable(d);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/person/v_plaunov/mylistview/CoinViewActivity.kt
package com.person.v_plaunov.mylistview

import android.database.sqlite.SQLiteDatabase
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.text.TextUtils
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream

class CoinViewActivity : AppCompatActivity() {
    private var myDataBase: SQLiteDatabase? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_coinview)
        val intent = intent
        val bundle = intent!!.extras
        val coinId = bundle!!.getInt("coin_id")
        //        String coinState = bundle.getString("coin_state");
//        String coinNominal = bundle.getString("coin_nominal");
//        String coinImgName = bundle.getString("coin_img");

        // переменные для query
        val columns = arrayOf("_id", "Nominal", "State", "Img", "Year", "Theme", "Description")
        val selection: String
        val selectionArgs: Array<String>

        //coins.clear();
        //Наш ключевой хелпер
        val dbOpenHelper = DBOpenHelper(this, DB_NAME)
        myDataBase = dbOpenHelper.openDataBase()
        //Все, база открыта!

//        Cursor cursor = myDataBase.query("coins", columns, selection, selectionArgs, null, null, null);
        selection = "_id = ?"
        selectionArgs = arrayOf(coinId.toString())
        val cursor = myDataBase?.query("coins", columns, selection, selectionArgs, null, null, null)
        var coinImagePath = ""
        cursor!!.moveToFirst()
        val coinState = cursor.getString(cursor.getColumnIndexOrThrow("State"))
        val cState = findViewById<TextView>(R.id.coin_state)
        cState.text = coinState
        val coinNominal = cursor.getString(cursor.getColumnIndexOrThrow("Nominal"))
        val cNominal = findViewById<TextView>(R.id.coin_nominal)
        cNominal.text = coinNominal
        var coinImgName = cursor.getString(cursor.getColumnIndexOrThrow("Img"))
        if (!TextUtils.isEmpty(coinImgName)) coinImgName = coinImgName.replace("\\", "//")
        val coinYear = cursor.getString(cursor.getColumnIndexOrThrow("Year"))
        val cYear = findViewById<TextView>(R.id.coin_year)
        cYear.text = coinYear
        val coinTheme = cursor.getString(cursor.getColumnIndexOrThrow("Theme"))
        val cTheme = findViewById<TextView>(R.id.coin_theme)
        cTheme.text = coinTheme
        val coinDescr = cursor.getString(cursor.getColumnIndexOrThrow("Description"))
        val cDescr = findViewById<TextView>(R.id.coin_description)
        cDescr.text = coinDescr
        cursor.close()
        val packageName = this.packageName
        val DB_PATH = String.format(this.getString(R.string.str_db_path), packageName)
        if (!TextUtils.isEmpty(coinImgName)) coinImagePath = DB_PATH + coinImgName
        try {

            val inputStream: InputStream = FileInputStream(coinImagePath)
            val num = inputStream.available()
            //Drawable.createFromStream(inputStream, null)
            val d = Drawable.createFromPath(coinImagePath)

            val cImg = findViewById<TouchImageView>(R.id.coin_pic)
            cImg.setImageDrawable(d)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    companion object {
        //private static String DB_PATH = "/data/data/com.person.v_plaunov.mylistview/databases/";
        //private static String DB_PATH = "/storage/self/primary/My Documents/";
        private const val DB_NAME = "myDB"
    }
}